### PR TITLE
refactor(experiments): Using the derived hostname label for scheduling job on specific node via nodeSelector

### LINKED ISSUE #1344: feat(scheduling): Use nodeName to schedule the job on the specific node
**Description**

- In the present scenario, We are using node selector for scheduling the application on the specific node with label `kubernetes.io/hostname: <app_node>`  which is platform dependent. 

- The best way is to provide the node name in `spec.nodeName` so that it can directly schedule the resources on the specific node platform independently.

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

**What this PR does / why we need it**:

- Using nodeName for scheduling job on specific node

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1344 
**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: chaoslib/litmus/platform/gke/node-memory-hog-job.j2 [hunk 1]
```diff
@@ -16,8 +16,7 @@ spec:
         chaosUID: {{ chaos_uid }}
 {% endif %}
     spec:
-      nodeSelector:
-        kubernetes.io/hostname: {{ node_name }}
+      nodeName: {{ app_node }}
```
Comment (rajdas98) [inline chaoslib/litmus/platform/gke/node-memory-hog-job.j2]: app_node --> node_name

File: chaoslib/litmus/platform/gke/node-cpu-hog-job.j2 [hunk 2]
```diff
@@ -16,8 +16,7 @@ spec:
         chaosUID: {{ chaos_uid }}
 {% endif %}
     spec:
-      nodeSelector:
-        kubernetes.io/hostname: {{ node_name }}
+      nodeName: {{ app_node }}
```
Comment (rajdas98) [inline chaoslib/litmus/platform/gke/node-cpu-hog-job.j2]: app_node --> node_name

**Comment by ksatchit** [comment]
Looks great! Can we: 

- Have this run any one example on the supported platforms today just to make sure.
- Also what is the policy w/ spec.nodeName when there aren't enough resources ? Does the pod stay in pending state (as it seems to be the case with nodeSelector?) in which case it schedules once the resources are freed up? If the same behaviour shows up we are good to go.

Sample outputs for the above wil be appreciated!

**Comment by ispeakc0de** [comment]
**Container kill Job logs on gcp after changes**
[container-kill-gcp.txt](https://github.com/litmuschaos/litmus/files/4347747/container-kill-gcp.txt)

**Comment by rajdas98** [comment]
Hey shubham, Is it possible to run the litmus E2E pipeline using your image ?

**Comment by ispeakc0de** [comment]
> Hey shubham, Is it possible to run the litmus E2E pipeline using your docker image ? If yes can you run the pipeline

@rajdas98 No, Our e2e pipeline is not supporting this functionality yet

File: chaoslib/litmus/container_kill/containerd_chaos/crictl-chaos.yml [hunk 3]
```diff
@@ -39,14 +39,21 @@
       wait_for: timeout="{{ ramp_time }}"
       when: "ramp_time is defined and ramp_time != ''"
 
+    - name: Getting host name from the node
+      shell: >
+        kubectl get node {{ app_node }} -oyaml | grep hostname | awk -F ": " '{print $2}'
```
Comment (ksatchit) [inline chaoslib/litmus/container_kill/containerd_chaos/crictl-chaos.yml]:
```suggestion
        kubectl get node {{ app_node }} -o jsonpath='{.metadata.labels.kubernetes\.io/hostname}'
```

File: chaoslib/litmus/platform/gke/node_memory_consumption.yml [hunk 4]
```diff
@@ -39,6 +46,7 @@
       vars:
         stress_time: "{{ c_duration }}s"
         total_memory_percentage: "{{ memory_percentage }}%"
+        hsot_name: "{{ hostname.stdout }}"
```
Comment (ksatchit) [inline chaoslib/litmus/platform/gke/node_memory_consumption.yml]: host_name

**Pull request merged**